napalm_ios/ios.py

```diff
@@ -1439,6 +1439,104 @@ def ping(self, destination, source='', ttl=255, timeout=2, size=100, count=5):
 
         return ping_dict
 
+    def traceroute(self, destination, source='', ttl=0, timeout=0):
+        """
+        Executes traceroute on the device and returns a dictionary with the result.
+
+        :param destination: Host or IP Address of the destination
+        :param source (optional): Use a specific IP Address to execute the traceroute
+        :param ttl (optional): Maimum number of hops -> int (0-255)
+        :param timeout (optional): Number of seconds to wait for response -> int (1-3600)
+
+        Output dictionary has one of the following keys:
+
+            * success
+            * error
+
+        In case of success, the keys of the dictionary represent the hop ID, while values are
+        dictionaries containing the probes results:
+            * rtt (float)
+            * ip_address (str)
+            * host_name (str)
+        """
+
+        command = "traceroute {}".format(destination)
+        if source:
+            command += " source {}".format(source)
+        if ttl:
+            if isinstance(ttl, int) and 0 <= timeout <= 255:
+                command += " ttl 0 {}".format(str(ttl))
+        if timeout:
+            # Timeout should be intiger between 1 and 3600
+            if isinstance(timeout, int) and 1 <= timeout <= 3600:
+                command += " timeout {}".format(str(timeout))
+        # Have to extend "send_command timeout" since traceroute can last for long time
+        output = self.device.send_command_expect(command, max_loops=3000)
```

Comment:
@mirceaulinic I think that this will lead to the same "timeout" issue, that force me to use max_loops parameter... The cli method is using the same netmiko send_command method.

Maybe the problem is in test, because (if I understand that correctly) is using own implementation of the send_command method, which does not have the max_loops parameter...